tests/core_test.go

```diff
@@ -64,21 +78,44 @@ var _ = Describe("Core API endpoint tests", func() {
 			Expect(err).NotTo(HaveOccurred())
 		})
 
+		It("It should list resource using generic API client", func() {
+			// make sure at least one resource exists
+			ctx := context.Background()
+			createBackend(ctx, cli, nil)
+
+			genericAPI, err := api.NewAPI(api.WithClientOptions(client.AuthFromEnv(false)))
+			Expect(err).ToNot(HaveOccurred())
+			var pageIter types.PageInfo
+			err = genericAPI.List(ctx, &resource.Info{}, api.Paged(1, 100, &pageIter))
+			Expect(err).ToNot(HaveOccurred())
+			var resInfo []resource.Info
+			Expect(pageIter.Next(&resInfo)).To(BeTrue())
+			Expect(resInfo).ToNot(BeEmpty())
+			Expect(resInfo[0].Identifier).ToNot(BeEmpty())
+		})
+
+		It("It should throw an error for unsupported operations for the genric API client", func() {
+			// make sure at least one resource exists
+			ctx := context.Background()
+			createBackend(ctx, cli, nil)
```

Comment:
should this be a context?
```go
Context("with at least one backend existing", func() {
  JustBeforeEach(func() { createBackend() })
  It("...",...)
```

also, `It("should...", ...)`, not `It("It should ...", ...)`

Reply:
Like that very much